Libraries/dotNetRDF/Writing/GraphMLWriter.cs

```diff
@@ -0,0 +1,282 @@
+{
+    /// <summary>
+    /// Class for serializing a <see cref="IGraph">graph</see> in GraphML format
+    /// </summary>
+    public class GraphMLWriter : IStoreWriter
+    {
+        /// <summary>
+        /// Event raised when there is ambiguity in the syntax being producing
+        /// </summary>
+        /// <remarks>This class doesn't raise this event</remarks>
+        public event StoreWriterWarning Warning;
+
+        /// <summary>
+        /// Saves a triple store to a file in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="filename">The name of the target file</param>
+        public void Save(ITripleStore store, string filename)
+        {
+            using (var writer = new StreamWriter(File.OpenWrite(filename)))
+            {
+                this.Save(store, writer);
+            }
+        }
+
+        /// <summary>
+        /// Saves a triple store to a text writer in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="output">The target text writer</param>
+        public void Save(ITripleStore store, TextWriter output)
+        {
+            this.Save(store, output, false);
+        }
+
+        /// <summary>
+        /// Saves a triple store to a text writer in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="output">The target text writer</param>
+        /// <param name="leaveOpen">Boolean flag indicating if the output writer should be left open by the writer when it completes</param>
+        public void Save(ITripleStore store, TextWriter output, bool leaveOpen)
+        {
+            using (var writer = XmlWriter.Create(output, new XmlWriterSettings { CloseOutput = !leaveOpen, OmitXmlDeclaration = true }))
+            {
+                this.Save(store, writer);
+            }
+        }
+
+        /// <summary>
+        /// Saves a triple store to an XML writer in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="output">The target XML writer</param>
+        public void Save(ITripleStore store, XmlWriter output)
+        {
+            GraphMLWriter.WriteGraphML(output, store);
+        }
+
+        private static void WriteGraphML(XmlWriter writer, ITripleStore store)
+        {
+            writer.WriteStartElement(GraphMLHelper.GraphML, GraphMLHelper.NS);
+
+            GraphMLWriter.WriteKey(writer, GraphMLHelper.EdgeLabel, GraphMLHelper.Edge);
+            GraphMLWriter.WriteKey(writer, GraphMLHelper.NodeLabel, GraphMLHelper.Node);
+
+            foreach (var graph in store.Graphs)
+            {
+                GraphMLWriter.WriteGraph(writer, graph);
+            }
+
+            writer.WriteEndElement();
+        }
+
+        private static void WriteGraph(XmlWriter writer, IGraph graph)
```

Comment:
I'm worried that this can be not too performant with multiple loops over the graph. Have you tried it on larger datasets?